main.js

```
@@ -7,7 +7,20 @@ const _ = {
     }
     return newIterable;
   },
+  reduce(iterable, reducer, initialValue) {
+    if (typeof reducer !== 'function') {
```

If you're making `initialValue` optional then you will want to throw an error if `initialValue` is `undefined` but `iterable` is empty.  Do you see why?